server/storage/mvcc/revision.go

```diff
@@ -14,48 +14,84 @@
 
 package mvcc
 
-import "encoding/binary"
+import (
+	"encoding/binary"
 
-// revBytesLen is the byte length of a normal revision.
-// First 8 bytes is the revision.main in big-endian format. The 9th byte
-// is a '_'. The last 8 bytes is the revision.sub in big-endian format.
-const revBytesLen = 8 + 1 + 8
+	"go.uber.org/zap"
+)
 
-// A revision indicates modification of the key-value space.
-// The set of changes that share same main revision changes the key-value space atomically.
-type revision struct {
-	// main is the main revision of a set of changes that happen atomically.
-	main int64
+const (
+	// revBytesLen is the byte length of a normal revision.
+	// First 8 bytes is the revision.main in big-endian format. The 9th byte
+	// is a '_'. The last 8 bytes is the revision.sub in big-endian format.
+	revBytesLen = 8 + 1 + 8
+	// markedRevBytesLen is the byte length of marked revision.
+	// The first `revBytesLen` bytes represents a normal revision. The last
+	// one byte is the mark.
+	markedRevBytesLen      = revBytesLen + 1
+	markBytePosition       = markedRevBytesLen - 1
+	markTombstone     byte = 't'
+)
 
-	// sub is the sub revision of a change in a set of changes that happen
+// RevisionTombstone indicates modification of the key-value space.
+// The set of changes that share same main revision changes the key-value space atomically.
+type RevisionTombstone struct {
```

Comment:
Note, when we transfer revision from or to bytes in MVCC, we are always doing operation on RevisionTombstone. However, when we compare a revision, we always operate on only the Revision.

Have you thought about creating a sub-struct for revision? For example:
```
type RevisionTombstone struct {
  Revision
  Tombstone bool
}
type Revision struct {
  Main int64
  Sub int64
}
```

Reply:
Yeah I had considered it, but given the previous comments, I wasn't sure if that would be too tightly coupling the mvcc and versioning. Understanding that better now, I'll make this change